src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+        b_exp[1:4, 1:4] = ms_coefficients4
+        k = order # For implicit method, order = k
+        for s = 5:k
+            for j = 0:(s - 1)
+                # Assign in correct order for multiplication below
+                #  (a factor depending on j and s) .* (an integral of a polynomial with -(0:s), except -j, as roots)
+                p_int = Polynomials.polyint(Polynomials.poly(diagm(-[0:j - 1; j + 1:s - 1])))
+                b_exp[s, s - j] = ((-1)^j / factorial(j)
+                               / factorial(s - 1 - j) * Polynomials.polyval(p_int, 1))
+            end
+        end
+    end
+
+    dt = diff(t)
+    y = Array(typeof(y0), length(t))
+    dy = similar(y)
+    y[1] = y0
+    dy[1] = F(t[1],y[1])
+
+    ##PECE Method for Implicit Adams solver
+    for i=1:length(t)-1
+        steporder = min(i,order)
+
+        ##(P)redict y[i+1] using explicit Adams Bashforth coefficients
+        y[i+1]=y[i]
+        for j=1:steporder
+            y[i+1] += dt[i]*b_exp[steporder, j]*dy[i-(steporder-1) + (j-1)]
+        end
+
+        ##(E)valuate function F at the approximate point t[i+1],y[i+1]
+        dy[i+1] = F(t[i+1],y[i+1])
+
+        ##(C)orrect the formula using implicit Adams Moulton coefficients
+        y[i+1]=y[i]
+        for j = 1 : steporder
+            y[i+1] += dt[i]*b_imp[steporder, j]*dy[i - (steporder -1) + j]
+        end
+
+        ##(E)valulate the function anew at corrected approximatiion t[i+1],y[i+1]
+        dy[i+1] = F(t[i+1], y[i+1])
+    end
+    return t, y
+end
+
+ode_am5(F::Function,y0, t)= ode_am(F::Function,y0, t,5)
+
+
+################################################################################
+# Variable Step Variable Order (VSVO) Adams-Moulton Solver for nonstiff problems
+#
+#     [T,X] = ODE113(ODEFUN, Y0, TSPAN,ORDER=4;kwords...) with TSPAN = [T0:TFINAL]
+#     or[T0,T1,...,TFINAL] for dense output
+#
+#     (Main ref: Hairer & Wanner 1996, Vol I, p.356-360, p.398-400, p.421-423)
+################################################################################
+function ode113(F, y0, tspan, order::Integer=4; kwords...)
+    # For y0 which don't support indexing.
+    F_ = (t, y) -> [F(t, y[1])]
+    t,y = ode113(F_, [y0], tspan, order; kwords...)
+    return t, vcat_nosplat(y)
+end
+
+"a variable stepsize, variable order multi-step solver for nonstiff ODEs"
+function ode113(F, y0::AbstractVector, tspan, order::Integer=4;adaptive_order = true,
+                                                      max_order = 13,
+                                                      reltol = 1.0e-9,
+                                                      abstol = 1.0e-9,
+                                                      minstep=abs(tspan[1] - tspan[end])/1e18,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.5,
```

Comment:
Whilst I agree that those should become options at some point, probably shouldn't introduce new keywords in this PR.